pysr/regressor_sequence.py

```diff
@@ -0,0 +1,425 @@
+import os
+import pickle as pkl
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+from numpy.typing import NDArray
+from sklearn.base import BaseEstimator, MultiOutputMixin, RegressorMixin
+
+from .export_latex import (
+    sympy2latex,
+    sympy2latextable,
+    sympy2multilatextable,
+    with_preamble,
+)
+from .sr import PySRRegressor
+from .utils import (
+    ArrayLike,
+    PathLike,
+    _csv_filename_to_pkl_filename,
+    _preprocess_julia_floats,
+    _safe_check_feature_names_in,
+    _subscriptify,
+    _suggest_keywords,
+)
+
+
+def _check_assertions(
+    X,
+    recursive_history_length=None,
+    weights=None,
+    variable_names=None,
+    X_units=None,
+):
+    if recursive_history_length is not None and recursive_history_length <= 0:
+        raise ValueError(
+            "The `recursive_history_length` parameter must be greater than 0 (otherwise it's not recursion)."
+        )
+    if len(X.shape) > 2:
+        raise ValueError(
+            "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+        )
+    if len(X) <= recursive_history_length + 1:
+        raise ValueError(
+            f"Recursive symbolic regression with a history length of {recursive_history_length} requires at least {recursive_history_length + 2} datapoints."
+        )
+    if isinstance(weights, np.ndarray) and len(weights) != len(X):
+        raise ValueError("The length of `weights` must have shape (n_times,).")
+    if isinstance(variable_names, list) and len(variable_names) != X.shape[1]:
+        raise ValueError(
+            "The length of `variable_names` must be equal to the number of features in `X`."
+        )
+    if isinstance(X_units, list) and len(X_units) != X.shape[1]:
+        raise ValueError(
+            "The length of `X_units` must be equal to the number of features in `X`."
+        )
+
+
+class PySRSequenceRegressor(MultiOutputMixin, RegressorMixin, BaseEstimator):
```

Comment:
What do `MultiOutputMixin` and `RegressorMixin` add to the methods? I'm not sure it's compatible with them since they have different signatures.

Reply:
oh true this is wrapper of PySRRegressor